### DISCUSSION TITLE
Need of a small tutorial on how to write a consistent and basic article website

### DISCUSSION THREAD

**Discussion by ysig** [mmistakes/minimal-mistakes#2949] [Q&A]
Hi, 
thanks a lot for this awesome repo and this really motivating community work.

## Motivation

I have recently started creating my website on https://ysig.github.io/ and in https://github.com/ysig/ysig.github.io

I have the following problem:

I have created subfolders in order to put markdown files in order to post articles (as in [_data/navigation.yml](https://github.com/ysig/ysig.github.io/blob/master/_data/navigation.yml)).

My problems are the following and were not resolved from the [quick start guide](https://mmistakes.github.io/minimal-mistakes/docs/quick-start-guide/):

- How can the recent posts be automatically or manually updated with recent updates on those directories?
Right now my recent posts column looks like this:
![image](https://user-images.githubusercontent.com/28439529/117566909-0214e580-b0c2-11eb-97d6-98ef28cdf5dc.png)

- Moreover I observe a lack of spacing and centering in the subfolders I have.
So for a example the main page is well spaced:
![image](https://user-images.githubusercontent.com/28439529/117566740-29b77e00-b0c1-11eb-9f73-648c97cd6c79.png)
whereas when I click one of the tabs:
![image](https://user-images.githubusercontent.com/28439529/117566758-3e941180-b0c1-11eb-8d94-e32169af7ab2.png)
I see this:
![image](https://user-images.githubusercontent.com/28439529/117566771-4f448780-b0c1-11eb-8083-67ae1ba8f87f.png)
i.e. there is no spacing or centering.
Why is this happening? Should I add a command on the begging of my markdown file?

## Suggestion

I have not found any tutorial that can help me resolve this issue and thus I share it here, so that you maybe can add relevant information to your docs.
I would like to note, that although I am developer, I know nothing more than the absolute basics of web development.

I remain at your disposal for further questions.
Thanks a lot!!!

**Comment by iBug** [comment]
Looks like you're using the `default` layout when you're supposed to use [`single`](https://mmistakes.github.io/minimal-mistakes/docs/layouts/#single-layout) instead.

**Reply by ysig** [reply]
No I am not:

![Screenshot_20210509-174254_Chrome](https://user-images.githubusercontent.com/28439529/117578349-3b634a80-b0ee-11eb-9ca1-01d557e10ae7.jpg)

**Reply by iBug** [reply]
Apparently it didn't apply. Your scope `type: posts` matches nothing in your repository:

![image](https://user-images.githubusercontent.com/7273074/117581223-55bd1880-b12e-11eb-97c0-b0d55e913dc1.png)

Look at [Jekyll documentation](https://jekyllrb.com/docs/step-by-step/09-collections/#front-matter-defaults), `type: posts` only applies to documents under `_posts` directory, which is treated as the "posts" collection.

If you want a `defaults:` to apply to everything in your repository, you should drop the `type:` key from scope.

**Reply by ysig** [reply]
Your second approach fixed it.
But still recent posts is empty and I also see this:
![image](https://user-images.githubusercontent.com/28439529/117581929-41662500-b108-11eb-8231-07d0d0674828.png)
Do you know why the title appears twice?
Thank you!

**Reply by iBug** [reply]
"Recent posts" only fetches posts from the "posts" collection, i.e. you need to put them under the `_posts` directory.

Also don't start your *content* with an H1 (like `# Heading 1`) as it'll be repeated. Only specify the title in front matter and don't repeat it.

**Reply by ysig** [reply]
Hi! Thank you for your answer:  

->"Recent posts" only fetches posts from the "posts" collection, i.e. you need to put them under the _posts directory.
When I moved everything under posts and changed my navigation to /_posts/.. the files where not reachable and I got a 404 error.

-> Also don't start your content with an H1 (like # Heading 1) as it'll be repeated. Only specify the title in front matter and don't repeat it.
What do you mean front matter? If it is jekyll I don't understand how to do it as all my files are in .md

Thank you!

**Reply by iBug** [reply]
Read Jekyll documentation carefully, everything is covered:

- [Collections](https://jekyllrb.com/docs/collections/) (for `_posts` directory)
- [Front Matter](https://jekyllrb.com/docs/front-matter/)

**Reply by ysig** [reply]
I read a bit those documents but still I haven't fixed this problem and it even get's worse.
The posts directory is not visible on my website neither when looking under `_posts` or `posts`.
More over adding:
```
---
title: Texts
layout: single
---

```
on the begging of the [markdown](https://ysig.github.io/texts/) removes the previously fixed properties of the page: centering and more.
Could you please give me detailed instruction of what I should do, as it's not at all obvious?

**Reply by iBug** [reply]
Oh I forgot to mention, Markdown source files under `_posts` directory must be named with date, e.g. `2021-02-31-example.md` or they won't be picked up by Jekyll. See [Posts](https://jekyllrb.com/docs/posts/) from Jekyll documenation. After that your Relates Posts should be populated.

For the layout issue, [you're still not](https://github.com/ysig/ysig.github.io/blob/8fdee000cfa167b016110ede87ec52e91bd1f80e/texts/index.md) using the right layout (you specified `layout: post` which is non-existent and falling back to `layout: default`). Check carefully.

**Reply by ysig** [reply]
For the first. Ok I see. But then how should I access them in navigation?
as /_posts/texts.md ?

Secondly can you tell me which layout should I use? Ideally I would just like to specify the title?

Thank you,

**Reply by iBug** [reply]
Depends on your `permalink` setting. See [Permalink](https://jekyllrb.com/docs/permalinks/).

**Reply by ysig** [reply]
Can you please help me?
I don't understand where I should specify the permalink?
On the begging of each file?
The global permalink is

`permalink: /:categories/:title/`

this doesn't make sense to me as I don't understand what categories should be.

Thanks

**Reply by iBug** [reply]
Go find some examples and compare with your repository, for example [mm-github-pages-starter](https://github.com/mmistakes/mm-github-pages-starter) or [the `docs/` folder](https://github.com/mmistakes/minimal-mistakes/tree/master/docs) of this repo. Consult [Jekyll documentation](https://jekyllrb.com) otherwise.

You need to learn by yourself, not just keep asking.

**Reply by ysig** [reply]
I don't have any problem learning but myself, but I am not making a website from scratch using Jekyll, I am using your template.
Ok I will look at the `mm-github-pages-starter` (this is what I was asking from the beggining).
Last question. Is it possible to have some posts without a date?

Thank you.
